schedule.go

```diff
@@ -0,0 +1,187 @@
+	UTCOffset     string `json:"utc_offset"`
+	First         int    `json:"first"`
+	After         string `json:"after"`
+}
+
+// GetScheduleResponse is the response data in GetSchedule
+type GetScheduleResponse struct {
+	ResponseCommon
+
+	Data GetScheduleData
+}
+
+type GetScheduleData struct {
+	Schedule   ScheduleData          `json:"data"`
+	Pagination GetSchedulePagination `json:"pagination"`
+}
+
+type ScheduleData struct {
+	Segments         []GetScheduleSegment `json:"segments"`
+	BroadcasterID    string               `json:"broadcaster_id"`
+	BroadcasterName  string               `json:"broadcaster_name"`
+	BroadcasterLogin string               `json:"broadcaster_login"`
+	Vacation         GetScheduleVacation  `json:"vacation"`
+}
+
+type GetScheduleSegment struct {
+	ID            string                     `json:"id"`
+	StartTime     string                     `json:"start_time"`
+	EndTime       string                     `json:"end_time"`
+	Title         string                     `json:"title"`
+	CanceledUntil string                     `json:"canceled_until"`
+	Category      GetScheduleSegmentCategory `json:"category"`
+	IsRecurring   bool                       `json:"is_recurring"`
+}
+
+type GetScheduleSegmentCategory struct {
+	ID   string `json:"id"`
+	Name string `json:"name"`
+}
+
+type GetScheduleVacation struct {
+	StartTime string `json:"start_time"`
+	EndTime   string `json:"end_time"`
+}
+
+type GetSchedulePagination struct {
+	Cursor string `json:"cursor"`
+}
+
+// Gets the broadcaster’s streaming schedule.
+// You can get the entire schedule or specific segments of the schedule
+func (c *Client) GetSchedule(params *GetScheduleParams) (*GetScheduleResponse, error) {
+	resp, err := c.get("/schedule", &GetScheduleData{}, params)
+	if err != nil {
+		return nil, err
+	}
+
+	schedule := &GetScheduleResponse{}
+	resp.HydrateResponseCommon(&schedule.ResponseCommon)
+	schedule.Data.Schedule = resp.Data.(*GetScheduleData).Schedule
+	schedule.Data.Pagination = resp.Data.(*GetScheduleData).Pagination
+
+	return schedule, nil
+}
+
+type UpdateScheduleParams struct {
+	BroadcasterID     string `json:"broadcaster_id"`
+	IsVacationEnabled bool   `json:"is_vacation_enabled"`
+	VacationStartTime string `json:"vacation_start_time"`
+	VacationEndTime   string `json:"vacation_end_time"`
+	Timezone          string `json:"timezone"`
+}
+
+type UpdateScheduleResponse struct {
+	ResponseCommon
+}
+
+// Updates the broadcaster’s schedule settings, such as scheduling a vacation
+func (c *Client) UpdateSchedule(params *UpdateScheduleParams) (*UpdateScheduleResponse, error) {
+	resp, err := c.get("/schedule/settings", nil, params)
+	if err != nil {
+		return nil, err
+	}
+
+	schedule := &UpdateScheduleResponse{}
+	resp.HydrateResponseCommon(&schedule.ResponseCommon)
+
+	return schedule, nil
+}
+
+type CreateScheduleSegmentParams struct {
+	BroadcasterID string `json:"broadcaster_id"`
+	StartTime     string `json:"start_time"`
```

Comment:
`StartTime` should be of type [`Time`](https://github.com/nicklaw5/helix/blob/main/time.go#L24), rather than `string`.